File: lang/go/test/fun/vac/algorithms/search/BinarySearch/BinarySearchTest.go
package main

import (
    "fun/vac/algorithms/search/BinarySearch"
    "fun/vac/util/SequenceBuilder"
    "fun/vac/util/TestRunner"
)

func testBinarySearch() bool {
    size := 32768

    arr := make([]int, size)
    SequenceBuilder.PackIncreasing(arr)

    if BinarySearch.Find(arr, -1) != size {
        return false
    }

    if BinarySearch.Find(arr, 2147483647) != size {
        return false
    }

    for i, v := range arr {
        if BinarySearch.Find(arr, v) != i {
            return false
        }
    }

    return true
}

func main() {
    TestRunner.ParseTest(testBinarySearch())
}

File: archive/c++/muse/algorithms/search/BinarySearchTest.cpp
#include <muse/algorithms/search/BinarySearch.hpp>
#include <muse/util/SequenceBuilder.hpp>
#include <muse/util/TestRunner.hpp>

using namespace std;

bool testBinarySearch() {
  size_t size = 32768;

  vector<int> arr(size);
  SequenceBuilder::packIncreasing(arr);

  if (BinarySearch::find(arr, -1) != size) {
    return false;
  }

  if (BinarySearch::find(arr, 2'147'483'647) != size) {
    return false;
  }

  for (size_t i = 0; i < size; i++) {
    if (BinarySearch::find(arr, arr[i]) != i) {
      return false;
    }
  }

  return true;
}

int main() { TestRunner::pick(&testBinarySearch); }
